File: Santiago-Villarreal/Programacion/Java/TercerSemestre/Leccion 1/MundoPC/src/ar/com/system2023/MundoPC/Raton.java

package ar.com.system2023.MundoPC;

public class Raton extends DispositivosEntrada{
    private final int idRaton;
    private static int contadorRatones;
    
    public Raton (String tipoEntrada, String marca){
        super(tipoEntrada, marca);
        this.idRaton = ++Raton.contadorRatones;
    }

    @Override
    public String toString() {
        return "Raton{" + "idRaton=" + idRaton + ","+ super.toString()+'}';
    }
    
    
}

File: GianniPasquinelli/phyton/leccion14/mundo_pc/Raton.py
from Dispositivo_entrada import Dispositivo_entrada

class Raton(Dispositivo_entrada):
  
  contador_ratones = 0

  def __init__(self, tipo_entrada, marca):
    Raton.contador_ratones += 1
    self._id_raton = Raton.contador_ratones
    super().__init__(tipo_entrada, marca)

  def __str__(self):
    return f"Id: {self._id_raton}, Marca: {self._marca}, Tipo de entrada: {self._tipo_entrada}"
  
  
# pruebas
if __name__ == '__main__':
  raton1 = Raton('USB', 'HP')
  print(raton1)
  raton2 = Raton('Bluetooth', 'Dell')
  print(raton2)
  raton3 = Raton('USB', 'Logitech')
  print(raton3)
  raton4 = Raton('Bluetooth', 'MSI')
  print(raton4)
  raton5 = Raton('USB', 'HP')
  print(raton5)